### ISSUE TITLE
Missing frame when using GIF in background an XML template

### ISSUE THREAD

**Issue by Bladehunt0** [pabletos/pizzoo#1]
As already written in Reddit: If you use a GIF as background and want to add text etc. via XML, the GIF flashes one frame in the background color before it loops. Changeing the loop=True or False has no affect. Here the code:

```
from pizzoo import Pizzoo, WindowRenderer, ImageRenderer
from time import sleep

pizzoo = Pizzoo('192.168.XXX.XX', debug=True)

pizzoo.cls()

pizzoo.draw_gif('Y:/OUR/PATH/sun1.gif', xy=(0, 0), size=(64, 64), loop=False)

pizzoo.render_template('''
    <pizzoo>
        <rectangle x="0" y="0" width="100%" height="100%" color="0" filled="false">
            <section x="0" y="0">
                <date format="DD" x="1" y="1" color="#ffffff" font="small"></date>
            </section>
        </rectangle>
    </pizzoo>
''')
```

The GIF is attached.

I'm working around it with "set_dial", it seems to work, but isn't as comfortable as XML ;-) .
If you have any questions or need a vid or something, let me know.

Cheers!

![sun1](https://github.com/user-attachments/assets/c1552420-b452-4a03-a618-73a7db1fc9ec)

**Comment by pabletos** [comment]
Thanks for your report!

As I suspected, the problem is by the fact that templates were thought, at first, with the idea of generating a single frame, as here is used in conjunction with a gif it wrongly set the first frame instead of adding it's info to all the frames set by the gif. So my solution will be adding a `background` property on the `<pizzoo>` node that can be a path or color, so it can set a whole color, image or gif to the background. 

This also will mean modifiying a lil bit the template render logic, so template rendering will be applied to every frame on the buffer, instead to the first one. 

I'll get to it as soon as I have some hours free to work on it 😁

**Comment by pabletos** [comment]
@Bladehunt0 Bug is fixed, you can download the new 0.9.12 version from Pypi. In your case, making it work with the new syntax is making this minor change:

```
from pizzoo import Pizzoo, WindowRenderer, ImageRenderer
from time import sleep

pizzoo = Pizzoo('192.168.XXX.XX', debug=True)

pizzoo.render_template('''
    <pizzoo background="Y:/OUR/PATH/sun1.gif">
        <section x="0" y="0">
            <date format="DD" x="1" y="1" color="#ffffff" font="small"></date>
        </section>
    </pizzoo>
''')
```

And it should render it on the most deep layer (As a background) 😁

**Comment by Bladehunt0** [comment]
Seems to work now, thank you!
